### PR TITLE
fix(theme-blog): multiple elements have the same id

### PR DESCRIPTION
**Pull request by numToStr**
Removes the following warning in Elements tab in chromium

> id attribute value must be unique: Document has multiple static elements with the same id attribute

---

~Q) Should I remove the `const slug` and just use `props.id` as slug value in a follow-up PR?~ I removed the `const slug` because of https://github.com/shuding/nextra/pull/481#discussion_r912340686

### CODE REVIEW

**Comment by shuding** [comment]
The anchor and id here is necessary so you can have this scroll to anchor effect: https://shud.in/posts/metaballs-1#shaders

File: packages/nextra-theme-blog/src/mdx-theme.tsx [hunk 1]
```diff
@@ -45,7 +45,6 @@ const HeaderLink = ({
   const slug = slugger.slug(ReactDOMServer.renderToStaticMarkup(children))
   return (
     <Tag {...props}>
-      <span className="subheading-anchor" id={slug} />
       <a href={'#' + slug} className="subheading">
```
Comment (numToStr) [inline packages/nextra-theme-blog/src/mdx-theme.tsx:48]: One thing I noticed that `slug` variable changes every time page reload after change but `props.id` stays the same.

**Comment by numToStr** [comment]
I have tested this change locally and scroll anchor seems to be working fine.

**Comment by shuding** [comment]
The idea of that anchor element was to position the focus a little bit above the title, so it will not be too close to the edge of the viewport (doesn't look good).

**Comment by numToStr** [comment]
> The idea of that anchor element was to position the focus a little bit above the title

If this was the idea then it's not working. Currently it anchors the heading at the top, instead of anchoring the `span` element (which I removed).

**Comment by B2o5T** [comment]
closing since this issues was fixed in this PR https://github.com/shuding/nextra/pull/536

**Pull request closed without merging**